agent/acs/handler/refresh_credentials_handler.go

```diff
@@ -160,6 +165,15 @@ func (refreshHandler *refreshCredentialsHandler) handleSingleMessage(message *ec
 		}
 		if roleType == credentials.ExecutionRoleType {
 			task.SetExecutionRoleCredentialsID(aws.StringValue(message.RoleCredentials.CredentialsId))
+			// Refresh domainless gMSA plugin credentials
+			if task.RequiresDomainlessCredentialSpecResource() {
```

Comment:
How about we perform this check in the setter function (and maybe rename it)? That way the refresh handler would be simpler and have one less responsibility.

Also I think `TestHandleRefreshMessageAckedWhenCredentialsUpdatedDomainlessGMSA` would become much simpler as we will be able to split it into two tests - 
1. Acks are sent when domainless gMSA setter does not fail (we might even be able to use an existing test for this)
2. Setter function for windows does not attempt to update registry if task does not require domainless GMSA.

Reply:
Addressed both. 

Item 1 is in Function TestHandleRefreshMessageAckedWhenCredentialsUpdated


Item 2 is in Function TestHandleRefreshMessageAckedAndNoSetDomainlessCredentialsWhenNonDomainlessGMSATask